_posts/2017-12-05-book-review-docker-up-running.md

```diff
@@ -0,0 +1,117 @@
+enough for getting an overview of Docker containers and image. Next chapters are
+focused more on best practices to setup the Docker engine. To be frank I was not
+aware of ways to debug the container at runtime. Nor about strategies for
+logging it. Even not on monitoring containers and a host machine. This book
+points few expected production glitches that you should keep in mind. I didn't
+like the depicted testing workflow by authors. I will look for some other
+references which highlight more strategies to construct your test workflow. If
+you are aware of any please share them with me via e-mail. I know about
+achieving auto-scaling using various orchestration tools. This book provides
+step by step guidance on configuring and using them. Mentioned tools are Docker
+Swarm, Centurion and Amazon EC2 container service. I am confused why the book is
+missing Kubernets and Helios here. As a part of advanced topics, you will find a
+comparison of various filesystems with a shallow overview of how Docker engine
+interacts with them. The same chapter is discussing available execution drivers
+and introduces LXC as another container technology. This API option is
+deprecated by Docker version 17.0.9 which makes libcontainer as the only
+dependency. I got to aware of how Docker containers are providing the
+virtualization layer using Namespaces. Docker limits the execution of container
+using CGroups (Control Groups). Namespaces and CGroups are GNU/Linux level
+dependencies used by Docker under the hood. If you are an API developer, then
+you should not skip Chapter 11. This chapter is discussing two well-followed
+patterns Twelve-Factor App and The Reactive manifesto. These guidelines are
+helpful while designing the architecture of your services. The book concludes
+with further challenges of using Docker as a container tool.
+
+One typo I found at page number 123, second last line, 'expore some of the
+tools...', here 'expore' is a typo and it should be 'explore some of the
+tools...'. I have submitted it to the official errata. At the time of writing
+this post it has not confirmed by the authority. It seems errata requests are
+not frequently reviewed. Hope responsible will confirm it soon.
+
+
+Who should read this book?
+
+* If you are developer who is unaware of the Docker or who is using Docker but
+  not aware of the architecture of it then I must advise to read this book.
+
+* If your job is to setup deployment cluster using Docker then this book can
+  help you to get informed about the Linux host internals. It will be helpful to
+  get aware of various debugging strategies and security guidelines for
+  container infrastructure. Good to get aware of how Docker engine is
+  communicating with your host workstation.
+
+* This book is not a reference book. If you already well aware of Docker, then I
+  will not advise you to not read this book. In such case, Docker documentation
+  can be the best reference.
+
```

Comment:
This is not a reference book. If you alrady know Docker well, then this book is not useful to you. In that case, the documentation is the best reference.

Reply:
Taking the liberty:

If you already know Docker well, then this book will not be useful.....